metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/usage.py

```diff
@@ -173,99 +326,150 @@ def __init__(self, config: BigQueryV2Config, report: BigQueryV2Report):
         self.config: BigQueryV2Config = config
         self.report: BigQueryV2Report = report
 
-    def add_config_to_report(self):
-        self.report.query_log_delay = self.config.usage.query_log_delay
-
     def _is_table_allowed(self, table_ref: Optional[BigQueryTableRef]) -> bool:
         return (
             table_ref is not None
             and self.config.dataset_pattern.allowed(table_ref.table_identifier.dataset)
             and self.config.table_pattern.allowed(table_ref.table_identifier.table)
         )
 
-    def generate_usage_for_project(
-        self, project_id: str, tables: Dict[str, List[str]]
+    def run(
+        self, projects: Iterable[str], table_refs: Collection[str]
     ) -> Iterable[MetadataWorkUnit]:
-        aggregated_info: Dict[
-            datetime, Dict[BigQueryTableRef, AggregatedDataset]
-        ] = collections.defaultdict(dict)
+        events = itertools.chain.from_iterable(
+            self._get_usage_events(project) for project in projects
+        )
```

Comment:
Ah I can just change how we get usage events then

Reply:
Actually super annoying because we set `bq_client = BigQueryClient(project=project_id)` for each project currently, and I don't know if this has an impact on things. Just going to hope it doesn't, because it feels like if I don't merge this soon i'm never going to merge it